source/distributed/tria.cc

```diff
@@ -2808,6 +2832,32 @@ namespace parallel
                 && cell->level_subdomain_id() != this->locally_owned_subdomain())
               this->number_cache.level_ghost_owners.insert(cell->level_subdomain_id());
 
+#ifdef DEBUG
+          // Check that level_ghost_owners is symmetric by sending a message
+          // to every one
+          {
+
+            MPI_Barrier(this->mpi_communicator);
+
+            std::vector<MPI_Request> requests;
+            for (std::set<unsigned int>::iterator it = this->number_cache.level_ghost_owners.begin();
+                 it != this->number_cache.level_ghost_owners.end();
+                 ++it)
+              {
+                int dummy = 0;
+                requests.push_back(MPI_Request());
+                MPI_Isend(&dummy, 1, MPI_INT, *it, 9001, this->mpi_communicator, &requests.back());
+                requests.push_back(MPI_Request());
+                MPI_Irecv(&dummy, 1, MPI_INT, *it, 9001, this->mpi_communicator, &requests.back());
```

Comment:
While we're there: since you're already sending an integer, is there any useful payload you could put into it? The entire effort here will be spent on the communication, so any time you spend on computing a piece of data to send, and on the other side to verify some assertion, is going to cost you nothing.